work13/index.js

```diff
@@ -0,0 +1,135 @@
+"use strict";
+const wrap = document.getElementById("js_wrap");
+const ul = document.getElementById("js_list");
+const modalBtn = document.getElementById("js_modalButton");
+const modal = document.getElementById("js_modal");
+
+const url = "https://mocki.io/v1/d6da0b8a-3546-419e-aaec-05f3a247c6d0";
+// const url = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// const url = "https://myjson.dit.upm.es/api/bins/eu2f";
+
+// 下記は、myjson繋がらない時の固定値
+// const url = {
+//   "data": [
+//     {
+//       "a": "bookmark",
+//       "img": "img/1.png",
+//       "alt": "画像１",
+//       "text": "ブックマーク"
+//     },
+//     {
+//       "a": "message",
+//       "img": "img/2.png",
+//       "alt": "画像２",
+//       "text": "メッセージ"
+//     }
+//   ]
+// }
+
+async function getData() {
+  try {
+    const response = await fetch(url);
+    if (response.ok) {
+      const json = await response.json();
+      return json;
+    } else {
+      // .okでなければerrorを投げる
+      throw new Error(`Server request failed:${response.statusText}`);
+    }
+    // 下記は、固定値をそのままpromiseの返り値とする
+    // const json = url
+    // return json
+  } catch (e) {
+    //上記のthrowでエラーが出ていれば、ここでコンソールに表示し、
+    //エラーを投げる
+    console.error(e)
+    throw new Error(e);
+  }
+}
+
+async function getListData() {
+  let listData;
+  try {
+    listData = await getData();
+  } catch (e) {
+    wrap.textContent = `エラー内容:${e.message}`;
+  } finally {
+    removeLoading();
+  }
+  if (listData.data.length === 0) {
+    wrap.textContent = "data is empty";
+    return;
+  }
+  return listData;
+}
+
+function removeLoading() {
+  const loading = document.querySelector(".loading")
+  loading.remove();
+}
+
+function renderListElement({ data }) {
+  const fragment = document.createDocumentFragment();
+
+  data.forEach((value) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+
+    a.href = value.a;
+    a.textContent = value.text;
+    img.src = value.img;
+    img.alt = value.alt;
+
+    fragment.appendChild(li).appendChild(a).prepend(img);
+  });
+
+  ul.appendChild(fragment);
+}
+
+function loading() {
+  const div = document.createElement("div");
+  const img = document.createElement("img");
+  const container = document.querySelector(".container");
+  const body = document.querySelector("html body");
+
+  div.classList.add("loading");
+  img.src = "./img/loading-circle.gif"
+
+  div.append(img);
+  body.insertBefore(div, container);
+}
+
+const init = async () => {
+  loading();
+  const data = await getListData();
+  renderListElement(data);
+}
```

Comment:
Q
I think there are functions written in function() and functions written in arrow functions, do you use them differently?

Reply:
Originally I was trying to use "init ()" as an argument to another function.

As pointed out, it was more consistent to modify it to "function".
thank you.

commit: [eb9a77f](https://github.com/yasu-pro/js-handson/pull/15/commits/eb9a77fa9ca8063888b36821b3d56959750ce25f)

Reply:
I simply asked the question because I was curious if Yasu-san dares to use it differently.
By the way, I didn't have a clear answer about that either, so I looked it up.
So far, it seems that all we need to know is that "function() can be rewritten with the Arrow function" and I am good to go.
However, there actually seems to be a difference. I hope you can read this for reference. I learned a lot too!

[関数式とアロー関数の違い](https://typescriptbook.jp/reference/functions/function-expression-vs-arrow-functions)

Reply:
Thank you for the article.
I read the difference between the normal function and the arrow function.
Especially this part seems to be used in the future.
I learned a lot.